packages/components/filters/filters-list/src/filters-list.tsx

```diff
@@ -64,11 +61,36 @@ const CustomMenuList = ({
 );
 
 function FiltersList(props: TFiltersListProps) {
-  const [selectedFilters, setSelectedFilters] = useState<string[]>([]);
-  const persistedFilters = props.filters
-    .filter((filter) => filter.isPersistent)
+  /**
+   * persisted filters: always visible
+   * applied filters: filters for which values have been selected
+   * visible filters = persisted + applied
+   *
+   * visibleFiltersFromProps = filters visible based on props
+   * localVisibleFilters = filters actually visible in component currently
+   */
```

Comment:
@misama-ct I personally think that this approach to allowing `FiltersList` to recreate its' UI state based on props works well.

I would also be more than happy to consider an alternate approach if you have any in mind.

Reply:
No, let's go with this approach. I'm confident this implementation will work and once it does, we'll just have a look at the code to see if we see potential and/or need for improvement.